### DISCUSSION TITLE
withAuth Middleware not working on Vercel deployment

### DISCUSSION THREAD

**Discussion by thejessewinton** [nextauthjs/next-auth#5072] [Help]
I've got a really weird issue using `withAuth`; for some reason my local configuration works just fine, but when I deploy to Vercel it's not authorizing a user. This is what my middleware looks like:

```
import { withAuth } from 'next-auth/middleware';

const middleware = withAuth({
  pages: {
    signIn: '/sign-in',
  },
});

export default middleware;
``` 

It lets me run the sign in flow, but then keeps redirecting me to `sign-in`. Any idea what this could be? It's only doing this on Vercel.

**Comment by techied** [comment]
Probably related #4969

**Reply by thejessewinton** [reply]
> Probably related #4969

I did see this one; the solution for them was downgrading Next to v12.2.2 and I was already running 12.2.2, so no dice.
